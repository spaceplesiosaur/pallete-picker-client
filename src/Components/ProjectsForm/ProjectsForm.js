import React, { useState, useEffect } from 'react';
import './ProjectsForm.scss';

const ProjectsForm = ({ addNewProject }) => {

  const nameState = useState('')
  const name = nameState[0]
  const setName = nameState[1]

  const handleNameChange = (e) => {
    return setName(e.target.value)
  }

  const makeProject = () => {
    addNewProject({name: name})
    setName('')
  } 
  return (
    <section className="projects-name-form">
      <h2>Create New Project</h2>
      <div className="formBox">
        <label for="name">Name</label>
        <input className="formInput" type="text" name="name" value={name} onChange={handleNameChange}></input>
        <button onClick={makeProject} className="saveProjectButton">Save Project</button>
      </div>
    </section>
  )
}

export default ProjectsForm
