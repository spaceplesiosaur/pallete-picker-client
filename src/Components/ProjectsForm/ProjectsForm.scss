.projects-name-form {
    h2 {
        font-size: 2em;
    }
    .formBox {
        
    }
}