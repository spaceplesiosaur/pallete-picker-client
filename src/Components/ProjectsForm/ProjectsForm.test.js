import React from 'react';
import { shallow } from 'enzyme';
import ProjectsForm from './ProjectsForm'

describe('ProjectsForm', () => {
  let wrapper;
  let mockEvent;

  beforeEach(() => {
    wrapper = shallow(<ProjectsForm
        addNewProject={jest.fn()}
      />)

    mockEvent = {target: {value: "Gazebo"}}
  })

  it('should render with the correct properites', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
