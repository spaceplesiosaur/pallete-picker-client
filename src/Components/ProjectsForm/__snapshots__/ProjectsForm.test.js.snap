// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectsForm should render with the correct properites 1`] = `
<section
  className="projects-name-form"
>
  <h2>
    Create New Project
  </h2>
  <div
    className="formBox"
  >
    <label
      for="name"
    >
      Name
    </label>
    <input
      className="formInput"
      name="name"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      className="saveProjectButton"
      onClick={[Function]}
    >
      Save Project
    </button>
  </div>
</section>
`;
